### PR TITLE
Fixes #74 - Use --password-stdin for docker login

### LINKED ISSUE #74: Remove warning for docker login
**Is this a BUG REPORT or FEATURE REQUEST?**:
Feature request
**What happened?**:
When running the builder there is a warning to pass in the password through stdin
**What did you expect to happen?**:
Warning should not appear
**How do you reproduce it (as minimally and precisely as possible)?**:
Run the builder with credentials for the registry
**Anything else we need to know?**:

**Environment**:

### PR DESCRIPTION
**Pull request by sajayantony**
**Purpose of the PR:**

**Fixes #74   *[Refer closing issues using keywords](https://help.github.com/articles/closing-issues-using-keywords/)*:

**Notes/Details:**

```release-note
```

@Azure/azure-container-registry

### CODE REVIEW

File: pkg/commands/docker.go [hunk 1]
```diff
@@ -31,15 +32,19 @@ type dockerUsernamePassword struct {
...
+	c := make(chan error, 1)
+
+	go func() {
+		c <- runner.ExecuteCmd("docker", []string{"login", "-u", u.username, "--password-stdin", u.registry}, strings.NewReader(u.password+"\n"))
+	}()
+
+	select {
+	case err := <-c:
+		return err
+	case <-time.After(time.Second * 15): // Wait for upto 15 seconds
```
[the remark this answers is not in the document]
Comment (sajayantony) [reply pkg/commands/docker.go]: This accepts stdin which can block forever.

**Comment by sajayantony** [comment]
@northtyphoon @ehotinger  Please merge when you feel appropriate.

**Comment by ehotinger** [comment]
LGTM @northtyphoon feel free to merge once you loop back to this thread.

File: pkg/commands/docker.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (shhsu) [reply pkg/commands/docker.go]: login can take more than 15 seconds sometimes though

[the remark this answers is not in the document]
Comment (sajayantony) [reply pkg/commands/docker.go]: Moved that out to 30 seconds. I'm really hoping we just use the client library or move to jessfraz/img

**Comment by shhsu** [comment]
🚢

**Pull request merged**